pytosql.py

```diff
@@ -0,0 +1,88 @@
+import sys
+
+import pandas
+import sqlalchemy
+
+from midterm import df_processing
+
+
+def main():
+    db_user = "root"
+    db_pass = "x11docker"  # pragma: allowlist secret
+    db_host = "localhost"
+    db_database = "baseball"
+    connect_string = (
+        f"mariadb+mariadbconnector://{db_user}:{db_pass}@{db_host}/{db_database}"
+    )
+
+    sql_engine = sqlalchemy.create_engine(connect_string)
+
+    pitching = """
+                    SELECT
+                t.game_id
+                , CASE
+                    WHEN b.winner_home_or_away = 'A' THEN winner_home_or_away = 0
+                    ELSE b.winner_home_or_away = 1 END AS binary_winner
+                , (b.away_runs*9)/i.inning as away_earned_run_avg
+                , (b.home_runs*9)/i.inning as home_earned_run_avg
+                , (t.Strikeout+t.Walk)/i.inning as power_finesse
+                , ((t.stolenBase2B+t.stolenBase3B+t.stolenBaseHome)
+                /(t.caughtStealing2B+t.caughtStealing3B+t.caughtStealingHome)) as stolen_base_percentage
+                , t.Grounded_Into_DP AS double_play_opportunities
+                , t.Strikeout as strikeout_total
+                , CASE
+                    WHEN away_runs = 0 THEN 0
+                    WHEN home_runs = 0 THEN 1
+                    ELSE NULL END AS shutout_team
+                , (bc.Strikeout/bc.toBase) as strikeout_to_walk_ratio
+                , b.winddir as wind_direction
+                , bc.Hit/bc.atBat as batting_avg
+                , bc.Groundout/bc.Flyout as ground_fly_ratio
+                , (bc.Hit+bc.Walk+bc.Hit_By_Pitch)/(bc.atBat+bc.Walk+bc.Hit_By_Pitch+bc.Sac_Fly) as on_base_percentage
+            FROM game g
+            JOIN inning i ON g.game_id = i.game_id
+            JOIN team_batting_counts bc ON g.game_id = bc.game_id
+            JOIN boxscore b ON g.game_id = b.game_id
+            JOIN team_pitching_counts t ON g.game_id = t.game_id
+            GROUP BY t.game_id
+    """
```

Comment:
I am not able to run this SQL script.
"[2022-11-17 22:07:28] [22007][1292] (conn=177) Truncated incorrect DOUBLE value: 'H'" showed